src/operator/tensor/indexing_op.h

```diff
@@ -493,6 +493,125 @@ void BatchTakeOpForward(const nnvm::NodeAttrs& attrs,
   });
 }
 
+/*!
+ * \brief The parameters of the one_hot operator.
+ */
+struct OneHotParam : public dmlc::Parameter<OneHotParam> {
+  int depth;
+  double on_value;
+  double off_value;
+  int axis;
+  int dtype;
+  DMLC_DECLARE_PARAMETER(OneHotParam) {
+    DMLC_DECLARE_FIELD(depth)
+      .describe("The dimension size at dim = axis.");
+    DMLC_DECLARE_FIELD(on_value)
+      .set_default(1.0f)
+      .describe("The value assigned to the locations represented by indices.");
+    DMLC_DECLARE_FIELD(off_value)
+      .set_default(0.0f)
+      .describe("The value assigned to the locations not represented by indices.");
+    DMLC_DECLARE_FIELD(dtype)
+      .set_default(mshadow::kFloat32)
+      .add_enum("float32", mshadow::kFloat32)
+      .add_enum("float64", mshadow::kFloat64)
+      .add_enum("float16", mshadow::kFloat16)
+      .add_enum("uint8", mshadow::kUint8)
+      .add_enum("int32", mshadow::kInt32)
+      .describe("DType of the output");
+  }
+};
+
+inline void GetOneHotParams(const OneHotParam& param, int* depth, double* on_value,
+                            double* off_value, int* dtype) {
+  *depth = param.depth;
+  CHECK_GE(*depth, 0) << "Dimension size, depth, must be a non-negative integer";
+  *on_value = param.on_value;
+  *off_value = param.off_value;
+  *dtype = param.dtype;
+}
+
+inline bool OneHotOpShape(const nnvm::NodeAttrs& attrs,
+                          std::vector<TShape> *in_attrs,
+                          std::vector<TShape> *out_attrs) {
+  const OneHotParam& param = nnvm::get<OneHotParam>(attrs.parsed);
```

Comment:
I think that we can check the data validity by setting constraints like `.set_lower_bound`, ... (See the comment above).